tests/test_network.py

```diff
@@ -124,6 +134,162 @@ def test_add_objective(factory, client):
     assert dataset.objective_key == objective.key
 
 
+@pytest.mark.parametrize('asset_name', [
+    'objective',
+    'algo',
+    'aggregate_algo',
+    'composite_algo',
+])
+def test_metadata(factory, client, asset_name):
```

Comment:
Since this test is trying out multiple metadata values, we could parametrize the metadata as well.

There could be another layer of parametrization to specify what the input metadata value, what the output should and which error should be raised.

Here is an example of combining parametrization:
```python
@pytest.mark.parametrize('y', (1, 2))
@pytest.mark.parametrize('x', ('a', 'b'))
def test_foo(x, y):
    assert True
```
This will be executed in the following order:

* a-1
* a-2
* b-1
* b-2

See also for how to handle exceptions SubstraFoundation/substra-backend#189